ipysheet/pandas_loader.py

```diff
@@ -87,7 +87,25 @@ def from_dataframe(dataframe):
 
 
 def _extract_column(data, idx):
-    return [row[idx]['value'] for row in data]
+    import numpy as np
+    import pandas as pd
+
+    if len(data) == 0:
+        return np.array()
+
+    type = data[0][idx]['options'].get('type', 'text')
```

Comment:
Only valid if the entire column is of the same type... @maartenbreddels What do you think we should do in this case? Taking into account that a Pandas [DataFrames have one dtype per column](http://pandas.pydata.org/pandas-docs/stable/getting_started/10min.html#viewing-data)

Reply:
but dtype could be object, so shouldn't be a problem. I actually want to deprecate Sheet.data, since it's double the data, I'm not sure we should depend on it.

Reply:
But how do you think we should retrieve the data? If not through `data`? Through cells?

Reply:
Thinking about it, I am completely fine with deprecating the Sheet data. It would simplify the code a lot (and improve performances as we don't synchronize the entire sheet data for every single change). And it's definitely doable to retrieve the data through cells for the DataFrame export.

Reply:
One intermediate solution would be, rename 'data' to '_data' (+possible  docstring) to make sure  people do not rely on it. Your call, I'm fine with both.

> It would simplify the code a lot

I don't think at the frontend there will be simplifications.

Reply:
I'll make a test PR this week, removing data :)

Reply:
Great, you chose the right path :)